### ISSUE TITLE
keplerian_propagate returns NaN for orbits with inc = 0 (ecliptic-plane singularity)

### ISSUE THREAD

**Issue by ben-cassese** [ben-cassese/jorbit#62]
# Bug: `keplerian_propagate` returns NaN for circular orbits exactly in the ecliptic plane

(report generated by Claude)

## Summary

`keplerian_propagate` (and therefore `Particle.ephemeris` and `System.ephemeris` for Keplerian
particles) silently returns NaN RA/Dec whenever the input Cartesian state has **exactly zero
z-component in the ecliptic frame**. This occurs for any orbit with `inc = 0` that is
constructed via a `KeplerianState` and passed to either ephemeris method.

## Environment

- jorbit version: 1.2.0
- JAX x64 enabled: yes (`jax.config.update("jax_enable_x64", True)`)
- Python 3.13

## Steps to Reproduce

```python
import jax
jax.config.update("jax_enable_x64", True)
import jax.numpy as jnp
import numpy as np
from astropy.time import Time
from jorbit import Particle
from jorbit.utils.states import KeplerianState, CartesianState

# Template Keplerian particle (same pattern as the orbitflow simulator)
template_state = KeplerianState(
    semi=jnp.asarray([2.5]),
    ecc=jnp.asarray([0.1]),
    inc=jnp.asarray([5.0]),
    Omega=jnp.asarray([80.0]),
    omega=jnp.asarray([40.0]),
    nu=jnp.asarray([0.0]),
    acceleration_func_kwargs={},
    time=2451545.0,
)
particle = Particle(state=template_state, gravity="keplerian")

# Orbit with inc=0, ecc=0 (circular, exactly in ecliptic plane)
orbit_kep = KeplerianState(
    semi=jnp.asarray([1.74]),
    ecc=jnp.asarray([0.0]),
    inc=jnp.asarray([0.0]),   # <-- exact zero
    Omega=jnp.asarray([0.0]),
    omega=jnp.asarray([0.0]),
    nu=jnp.asarray([250.17]),
    acceleration_func_kwargs={},
    time=0.0,
)

epoch = Time(65758.7, format="mjd", scale="tdb")
epoch_jd = float(epoch.tdb.jd)  # ~2465759.2

# Set the epoch on the state (this is the pattern used for propagation)
state = orbit_kep.replace(time=epoch_jd)

obstimes = Time(
    [65747.358, 65747.430, 65747.502],
    format="mjd", scale="utc",
)
observer = jnp.array([
    [0.532, 0.750, 0.327],
    [0.531, 0.751, 0.327],
    [0.530, 0.752, 0.328],
])

# This returns NaN RA/Dec
coords = particle.ephemeris(times=obstimes, observer=observer, state=state)
print(coords)
# <SkyCoord (ICRS): (ra, dec) in deg
#     [(nan, nan), (nan, nan), (nan, nan)]>
```

**Contrast with a CartesianState constructed directly from a numpy array** (same orbit,
same epoch, same observation times) — this works:

```python
# Orbit in ICRS Cartesian: inc=0 in ecliptic means z_ICRS ≠ 0
theta = np.array([-0.591, -1.504, -0.652, 0.012, -0.004, -0.002])  # AU, AU/day
cart_state = CartesianState(
    x=jnp.asarray([theta[:3]]),
    v=jnp.asarray([theta[3:]]),
    time=epoch_jd,
    acceleration_func_kwargs={},
)
coords2 = particle.ephemeris(times=obstimes, observer=observer, state=cart_state)
print(coords2)
# <SkyCoord (ICRS): (ra, dec) in deg
#     [(240.23, -20.66), ...]>   # valid
```

## Root Cause

`Particle.ephemeris` (Keplerian branch) extracts `x, v` from the state and passes them to
`_keplerian_integrate`, which first converts to the ecliptic frame:

```python
x_ecl = icrs_to_horizons_ecliptic(x[None, :])
v_ecl = icrs_to_horizons_ecliptic(v[None, :])
positions_ecl, _ = keplerian_propagate(x_ecl, v_ecl, t0, times, GM)
```

When the state originates from a `KeplerianState` with `inc=0`, the Cartesian representation
has **z_ecl = 0 exactly**. After the ICRS→ecliptic round-trip the z-component is still
exactly 0.0 (the transformation is a pure rotation and the round-trip is exact in floating
point for this case).

Inside `keplerian_propagate`, this calls `cartesian_to_elements(x_ecl, v_ecl, gm)`. The
standard Cartesian-to-Keplerian conversion has a **singularity for orbits in the reference
plane** (inc = 0): the ascending-node computation involves `atan2(h_z, ...)` where
`h = x × v` and for a perfectly in-plane orbit `h_z = |h|`, but intermediate quantities
in the argument-of-periapsis / true-anomaly computation can divide by zero or produce NaN.

When the state originates from a numpy array (ICRS Cartesian), the ICRS→ecliptic conversion
introduces floating-point rounding of order 1e-9 AU, giving a tiny but non-zero `z_ecl`.
This tiny perturbation is sufficient to avoid the singularity in `cartesian_to_elements`.

## Diagnostic

```python
from jorbit.particle import icrs_to_horizons_ecliptic
from jorbit.utils.kepler import keplerian_propagate

x_from_kep  = state.to_cartesian().x.flatten()   # z_ecl = 0.0 exactly → NaN
x_from_numpy = jnp.array(theta[:3])               # z_ecl ≈ 3e-9     → valid

print(icrs_to_horizons_ecliptic(x_from_kep[None, :]))   # z component: 0.0
print(icrs_to_horizons_ecliptic(x_from_numpy[None, :])) # z component: ~3e-9
```

## Affected Code Paths

- `Particle.ephemeris(state=KeplerianState(...))` when `inc == 0`
- `Particle.ephemeris(state=CartesianState(...))` when the state came from
  `some_keplerian_state.to_cartesian()` with `inc == 0`
- `_keplerian_residuals` (same underlying call)
- `System.ephemeris(...)` when the `System` is initialized from particles whose state
  originated from a `KeplerianState` with `inc == 0` — `System.ephemeris` (keplerian branch)
  calls `_keplerian_system_ephem` → `_keplerian_system_integrate`, which vmaps
  `_keplerian_integrate`, the same function at the root of this bug

## Suggested Fix

In `keplerian_propagate` (or `cartesian_to_elements`), handle the degenerate `inc = 0`
(and `inc = π`) case explicitly. For a perfectly in-plane orbit, `Omega` is undefined and
can be set to 0; the true anomaly is then measured from the x-axis directly:

```
nu = atan2(y_ecl, x_ecl)  (when inc == 0)
```

Alternatively, add a small guard in `cartesian_to_elements` that clamps `z_ecl` away from
exactly zero (e.g. `z = jnp.where(jnp.abs(z) < 1e-15, 1e-15, z)`) before computing the
angular momentum cross product. This is a common practice in Keplerian solvers.

## Workaround (until patched)

Convert the Keplerian state to a numpy array and back before calling `ephemeris`.

**For `Particle.ephemeris`** — pass a `CartesianState` constructed from a numpy array as the
`state` override:

```python
theta = cartesian_theta_from_orbit(orbit, epoch)  # numpy array via ICRS
cart = CartesianState(
    x=jnp.asarray([theta[:3]]),
    v=jnp.asarray([theta[3:]]),
    time=epoch_jd,
    acceleration_func_kwargs={},
)
coords = particle.ephemeris(times=obstimes, observer=observer, state=cart)
```

**For `System.ephemeris`** — `System.ephemeris` has no `state` override parameter, so the
workaround must be applied upstream when constructing the `Particle` objects that go into
the `System`. Apply the same numpy round-trip to each particle's state before passing them
to `System(particles=[...])`.

In both cases, the ICRS→ICRS roundtrip is a no-op numerically, but when the state is later
rotated to the ecliptic frame the result has z_ecl ≈ O(1e-9) rather than exactly 0.
